apps/src/applab/applab.js

```diff
@@ -1897,27 +1897,43 @@ Applab.getIdDropdown = function (filterSelector) {
  * @private
  */
 Applab.getIdDropdownFromDom_ = function (documentRoot, filterSelector) {
-  var divApplabChildren = documentRoot.find('#divApplab').children();
-  var elements = divApplabChildren.toArray().concat(
-      divApplabChildren.children().toArray());
+  var elements = documentRoot.find('#designModeViz [id^="' + applabConstants.DESIGN_ELEMENT_ID_PREFIX + '"]');
 
   // Return all elements when no filter is given
   if (filterSelector) {
-    elements = $(elements).filter(filterSelector).get();
-  }
-
-  return elements
-      .sort(function (elementA, elementB) {
-        return elementA.id < elementB.id ? -1 : 1;
-      })
-      .map(function (element) {
-        return {
-          text: quote(element.id),
-          display: quote(element.id)
-        };
-      });
+    elements = elements.filter(filterSelector);
+  }
+
+  return elements.sort(byId).map(function (_, element) {
+    var id = quote(element.id.replace(new RegExp('^' + applabConstants.DESIGN_ELEMENT_ID_PREFIX), ''));
+    return {text: id, display: id};
+  }).get();
+};
+
+/**
+ * Returns a list of IDs currently present in the DOM of the current screen,
+ * including the screen, sorted by z-index.
+ */
+Applab.getIdDropdownForCurrentScreen = function () {
+  var screen = $('#designModeViz .screen').filter(function () {
+    return this.style.display !== 'none';
+  }).first();
+
+  var elements = screen.find('[id^="' + applabConstants.DESIGN_ELEMENT_ID_PREFIX + '"]').add(screen);
+
+  return elements.sort(byZIndex).map(function (_, element) {
+    return element.id.replace(new RegExp('^' + applabConstants.DESIGN_ELEMENT_ID_PREFIX), '');
+  }).get();
 };
 
+function byZIndex(a, b) {
```

Comment:
I'm not sure I follow.  The spec refers to "z-order":
![z-order](https://cloud.githubusercontent.com/assets/413693/12729943/16cbb282-c8de-11e5-9fe3-0af2180b3052.png)